### PR TITLE
chore: prepare for Firebase SDK version bumps

### PR DESCRIPTION
**Pull request by Salakar**
## Description

See individual commits for details of changes.

## Related Issues

Unblocks #5936 & App Check.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by russellwheatley — APPROVED** [review]
providing it passes tests, LGTM.

**Review by Ehesp — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
LGTM - clever leaving admob in but unhitching it like that. I'll see if that's possible with RNFB...

**Comment by Salakar** [comment]
> LGTM - clever leaving admob in but unhitching it like that. I'll see if that's possible with RNFB...

@mikehardy I had to go quite far back on the iOS version of the pod since AdMob hasn't been updated for a while to keep the build happy. I reckon RNFB will be ok with the [latest version](https://developers.google.com/admob/ios/rel-notes) however since it was kept in sync with Firebase core.

**Review by kroikie — APPROVED** [review]

File: packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingPlugin.java [hunk 1]
```diff
@@ -142,28 +140,20 @@ public void onReceive(Context context, Intent intent) {
...
+  private Task<Map<String, Object>> getToken() {
     return Tasks.call(
         cachedThreadPool,
         () -> {
-          String senderId =
-              arguments.get("senderId") != null
-                  ? (String) arguments.get("senderId")
-                  : Metadata.getDefaultSenderId(FirebaseApp.getInstance());
-          String token = FirebaseInstanceId.getInstance().getToken(senderId, "*");
+          String token = Tasks.await(FirebaseMessaging.getInstance().getToken());
```
Comment (mikehardy) [inline packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingPlugin.java:156]: This is potentially a big deal - needs testing to determine if deleteToken/getToken actually behave now, c.f. 

https://github.com/firebase/firebase-ios-sdk/issues/6433#issuecomment-693502927
https://github.com/invertase/react-native-firebase/issues/3714#issuecomment-721027986
https://github.com/invertase/react-native-firebase/issues/1510#issuecomment-717514608

Comment (mikehardy) [reply packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingPlugin.java:156]: For anyone checking this later, if that ever happens. I added a test for this scenario in react-native-firebase that exercised the underlying ios / android SDKs and their behavior appears correct now, this is not an issue (anymore) as far as I can tell. :+1:

**Pull request merged**